yarn-project/simulator/src/avm/journal/journal.ts

```diff
@@ -208,10 +218,14 @@ export class AvmPersistableStateManager {
     this.log.debug(
       `noteHashes(${contractAddress})@${noteHash} ?? leafIndex: ${leafIndex} | gotLeafValue: ${gotLeafValue}, exists: ${exists}.`,
     );
-    // TODO(8287): We still return exists here, but we need to transmit both the requested noteHash and the gotLeafValue
-    // such that the VM can constrain the equality and decide on exists based on that.
-    const path = await this.merkleTree.getSiblingPath(MerkleTreeId.NOTE_HASH_TREE, leafIndex.toBigInt());
-    this.trace.traceNoteHashCheck(contractAddress, gotLeafValue, leafIndex, exists, path.toFields());
+    if (this.doMerkleOperations) {
+      // TODO(8287): We still return exists here, but we need to transmit both the requested noteHash and the gotLeafValue
```

Comment:
Is this true? Why do we need to transmit the requested noteHash? Witgen can generate the user-requested noteHash and confirm whether it equals the gotLeafValue

Reply:
Yeah you're right, i think the comment and issue are probably outdated